os/arch/arm/Kconfig

```diff
@@ -288,6 +288,16 @@ config ARCH_DABORTSTACK
 		will be used during data abort handling. Recommended data abort stack
 		size is 1K.
 
+config REG_STACK_OVERFLOW_PROTECTION
```

Comment:
I think we can choose the stack protection way just one of below configs, REG or MPU.
How about making option for selecting just one?

Reply:
Added condition

Reply:
Even this is selected, MPU based function is in the list of menuconfig. How about using choice?

Reply:
done